### PR TITLE
feat: add plain text output

### LINKED ISSUE #110: Consider plain text output
In addition to the output being markdown, it would be a nice-to-have to be able to leave out the markdown markup entirely. This is, rendering the html to plain text output. My primary motivation is feeding such output to a search index and secondly sending it via email (to non-technical people), although markdown is already good enough.

### PR DESCRIPTION
**Pull request by harlan-zw**
### 🔗 Linked issue

Resolves #110

### ❓ Type of change

- [ ] 📖 Documentation
- [ ] 🐞 Bug fix
- [ ] 👌 Enhancement
- [x] ✨ New feature
- [ ] 🧹 Chore
- [ ] ⚠️ Breaking change

### 📚 Description

Adds `format: 'text'` output across the JS engine, Rust NAPI, edge WASM, and CLI (`--format`, `--text`). Text mode strips Markdown and HTML syntax while keeping block spacing, table cell separators, image alt text, and preformatted content, with identical output between the JS and Rust engines in both direct and streaming conversion.

The Rust API stays source-compatible: existing `HTMLToMarkdownOptions` literals still compile, and `html_to_text`, `html_to_format`, and `MarkdownStreamProcessor::new_with_format` expose text output explicitly. The experimental `@mdream/js/text` subpath was dropped after review since it duplicated rendering rules and diverged from the main API on real fixtures.

Now merged with latest main: `format` is threaded through the shared `resolve-options` module from #121, and wrap width uses the zero default from #113.

## Summary by CodeRabbit

- **New Features**
  - Added plain-text output mode selectable via `format`/`--format`/`--text`, including new conversion APIs and matching streaming/chunk behavior.
- **Bug Fixes**
  - Plain-text rendering now consistently omits Markdown-specific constructs (including frontmatter/fenced-code handling) while preserving readable spacing, separators, and text content; improved streaming decoding across UTF-8 boundaries and enhanced parsing for aliased tags and edge-case markup.
- **Documentation**
  - Updated usage examples and option/CLI docs for plain-text output.
- **Tests**
  - Added extensive plain-text unit and streaming consistency coverage.

### CODE REVIEW

File: packages/js/src/text.ts [hunk 1]
```diff
@@ -0,0 +1,339 @@
...
+    return result
+  }
+
+  function getTextChunk(): string {
+    const current = currentText(state).trimStart()
+    const chunk = current.slice(lastYieldedLength)
+    lastYieldedLength = current.length
+    if (state.buffer.length > 1)
+      replaceBuffer(state, current)
+    return chunk
```
[the remark this answers is not in the document]
Comment (harlan-zw) [reply packages/js/src/text.ts]: Addressed in 825f55a and b103370. The standalone renderer was removed. The canonical JavaScript and Rust text paths preserve leading whitespace when output begins inside `pre` while continuing to trim ordinary document-leading whitespace; batch and streaming tests cover it.

File: packages/js/src/parse-core.ts [hunk 2]
```diff
@@ -0,0 +1,1402 @@
...
+  if (!foundClose) {
+    return {
+      complete: false,
+      newPosition: position,
+      remainingText: htmlChunk.substring(position),
+    }
+  }
+
+  const tagName = htmlChunk.substring(tagNameStart, i).toLowerCase() as keyof typeof TagIdMap
+  const tagId = TagIdMap[tagName] ?? -1
```
[the remark this answers is not in the document]
Comment (harlan-zw) [reply packages/js/src/parse-core.ts]: Fixed in b103370. End-tag parsing now stops the name at HTML whitespace, including form feed, and scans quote-aware to the real `>`. Sync and streaming regressions cover `</script\n>`, `</p >`, quoted `>`, and chunk splits.

File: packages/js/src/parse-core.ts [hunk 3]
```diff
@@ -0,0 +1,1402 @@
...
+        if ((state.depthMap[TAG_DT] || 0) > 0 || (state.depthMap[TAG_DD] || 0) > 0) {
+          closeImpliedTo(state, DT_DD, DL_SCOPE_BOUNDARY, handleEvent)
+        }
+      }
+    }
+  }
+
+  // Track global tag occurrence for uniqueness
+  const currentTagCount = (result.attributes && result.attributes.id) ? undefined : (state.depthMap[tagId] || 0)
+  state.depthMap[tagId] = (currentTagCount || 0) + 1
```
[the remark this answers is not in the document]
Comment (harlan-zw) [reply packages/js/src/parse-core.ts]: Fixed in b103370. `depthMap` now always increments from the current built-in nesting depth, regardless of an `id` attribute. A nested blockquote regression asserts depths `[1, 2]`.

File: packages/js/src/text.ts [hunk 4]
```diff
@@ -0,0 +1,339 @@
...
+
+function processText(state: TextState, node: TextNode, previous: Node | undefined): void {
+  if (node.excludedFromMarkdown || !node.value)
+    return
+
+  if (shouldAddSpacingBeforeText(lastChar(state), previous, node.value))
+    node.value = ` ${node.value}`
+
+  const wrapWidth = state.options?.wrapWidth
+  push(state, wrapWidth ? wrapText(node.value, currentColumn(state), wrapWidth) : node.value)
```
[the remark this answers is not in the document]
Comment (harlan-zw) [reply packages/js/src/text.ts]: Addressed in 825f55a and b103370. The standalone `@mdream/js/text` renderer was removed. The canonical `format: "text"` path now skips wrapping, synthetic spacing, and indentation inside `pre`, with cross-engine batch and streaming coverage.

**Comment by harlan-zw** [comment]
@coderabbitai Both findings addressed in 2e8fac3: `options.cleanUrls === true` is now computed once in `resolveCleanConfig`, and `options.extraction` / `options.tagOverrides` are read into locals before use. Kept `for...in` over `Object.entries` intentionally: this repo prefers avoiding the intermediate array allocation on hot paths.
